### PR TITLE
Add unit test demonstrating attaching a listener to queries

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Unit test for https://github.com/civicrm/civicrm-packages/pull/180

Before
----------------------------------------
No test

After
----------------------------------------
Test

Technical Details
----------------------------------------
Unit test for https://github.com/civicrm/civicrm-packages/pull/180 pretty much stalled on lack of a test for a long time - this is that test. It will fail until that is merged

I think we can also use this method to resolve https://bugs.mysql.com/bug.php?id=68184 which is an issue where for us truncate is very slow & we have had to hack to DELETE FROM instead

Comments
----------------------------------------
@seamuslee001 @davejenx

### CODE REVIEW

File: tests/phpunit/CRM/Core/DAOTest.php [hunk 1]
```diff
@@ -475,4 +475,46 @@ public function testCloneDAO() {
...
+    $listener = function(\Civi\Core\Event\GenericHookEvent $e) {
+      $e->query = '/* Forgot trailing comment marker' . $e->query;
+    };
+    Civi::dispatcher()->addListener('civi.db.query', $listener);
+    try {
+      CRM_Core_DAO::executeQuery('SELECT * FROM civicrm_domain');
+    }
+    catch (PEAR_Exception $e) {
+      $this->assertEquals(
+        "SELECT * FROM civicrm_domain [nativecode=1064 ** You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '/* Forgot trailing comment markerSELECT * FROM civicrm_domain' at line 1]",
```
Comment (davejenx) [inline tests/phpunit/CRM/Core/DAOTest.php]: Just wondering if this check could be a bit brittle if the exact text of MySQL's error message changes, or MariaDB diverges from MySQL, or someone runs the test with MySQL in a different locale. Could look for substrings like "`nativecode=1064`" and "`'/* Forgot trailing comment markerSELECT * FROM civicrm_domain'`" instead?

**Comment by seamuslee001** [comment]
Jenkins re test this please

**Comment by eileenmcnaughton** [comment]
@seamuslee001 this can't pass until the other is merged

**Comment by eileenmcnaughton** [comment]
test this please

**Comment by eileenmcnaughton** [comment]
Note from the main pr- docs issue outstanding https://github.com/civicrm/civicrm-packages/pull/180

**Comment by colemanw** [comment]
Test passes.

**Pull request merged**